### PR TITLE
Replace Zstandard compression with Brotli for CLP compression config.

### PR DESCRIPTION
**Pull request by junhaoliao**
# References
This PR is to facilitate easier WebUI integration with compression jobs submission. 

When submitting compression jobs, the config is first msgpack encoded then compressed with Zstandard before writing into the SQL database. One of the challenges with using Zstandard in a JavaScript (JS) environment is the lack of native JS support for this compression algorithm. While there are Zstandard compression libraries available online, many of them rely on WebAssembly to run. However, [the Meteor framework used in this project does not support WebAssembly well](https://github.com/meteor/meteor/issues/11143), which presents a significant barrier to using these Zstandard libraries effectively within the project's ecosystem.

Compression benchmarks that compare Zstandard, Brotli, and multiple other compression algorithms were conducted on [hive-24hr](https://zenodo.org/records/7094921) paths (34,923 in total), across various dimensions such as compression level, compressed size, compression time, and decompression time. The benchmark results, available at [https://github.com/junhaoliao/compression-benchmark/blob/master/Combined.csv](https://github.com/junhaoliao/compression-benchmark/blob/master/Combined.csv), indicate that Brotli (Quality: 4) offers a similar compression ratio with Zstandard (Level: 3) with comparatively slower compression and decompression time. 
| Algorithm | Compression Level | Time (seconds) | Compressed Size (bytes) | Decompress Time (seconds) |
| - | - | - | - | - |
| Zstandard | 3 (default) | 0.002377033233642578 | 166718 | 0.0010442733764648438 |
| Brotli | 4 | 0.009986162185668945 | 160929 | 0.0015523433685302734 |
| ... | ... | ... | ... | ... |

Internal discussions concluded that the slower compression and decompression time is acceptable in this setting of compression job config submission. On the other hand, Node.js 14 includes [native APIs for Brotli compression](https://nodejs.org/docs/latest-v14.x/api/zlib.html#zlib_zlib_brotlicompresssync_buffer_options). This native support simplifies the integration process and ensures that the compression functionality can be utilized efficiently within the Meteor framework and the specified Node.js version.

# Description
1. Replace Zstandard compression with Brotli for CLP compression config.

# Validation performed
1. Built and started CLP Package.
2. Submitted compression job with `<clp-package>/sbin/compress.sh ~/samples/hive-24hr/i-00c90a0f/`.
3. Observed successful job submission and completion. e.g.,
   ```
   2024-03-11 04:38:18,286 [INFO] [/opt/clp/lib/python3/site-packages/clp_package_utils/scripts/native/compress.py] Compression job 1 submitted.
   2024-03-11 04:38:19,802 [INFO] [/opt/clp/lib/python3/site-packages/clp_package_utils/scripts/native/compress.py] Compressed 7.64MB into 280.65KB (27.87)
   2024-03-11 04:38:19,804 [INFO] [/opt/clp/lib/python3/site-packages/clp_package_utils/scripts/native/compress.py] Compression finished. Runtime: 0.851241s. Speed: 8.97MB/s.
   ```

### CODE REVIEW

**Comment by junhaoliao** [comment]
Since there are Python dependency updates and I had to run `poetry update` to update the `poetry.lock` files for components [components/clp-py-utils](https://github.com/y-scope/clp/tree/main/components/clp-package-utils) and [components/job-orchestration](https://github.com/y-scope/clp/tree/main/components/job-orchestration), do I also update [components/clp-py-utils/poetry.lock](https://github.com/y-scope/clp/blob/main/components/clp-py-utils/poetry.lock) to avoid version mismatches in the CLP Package?

**Comment by kirkrodrigues** [comment]
@junhaoliao Yeah, let's update all the lock files.

**Review by haiqi96 — COMMENTED** [review]
The change to the package looks good to me,
Will let Kirk to be the gate keeper for poetry file changes

File: components/clp-package-utils/clp_package_utils/scripts/native/compress.py [hunk 1]
```diff
@@ -103,13 +103,11 @@ def handle_job_update(db, db_cursor, job_id, no_progress_reporting):
...
 def handle_job(sql_adapter: SQL_Adapter, clp_io_config: ClpIoConfig, no_progress_reporting: bool):
-    zstd_cctx = zstd.ZstdCompressor(level=3)
-
     with closing(sql_adapter.create_connection(True)) as db, closing(
         db.cursor(dictionary=True)
     ) as db_cursor:
         try:
-            compressed_clp_io_config = zstd_cctx.compress(
+            compressed_clp_io_config = brotli.compress(
                 msgpack.packb(clp_io_config.dict(exclude_none=True, exclude_unset=True))
```
Comment (kirkrodrigues) [inline components/clp-package-utils/clp_package_utils/scripts/native/compress.py:111]:
```suggestion
                msgpack.packb(clp_io_config.dict(exclude_none=True, exclude_unset=True)), quality=4
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

**Review by kirkrodrigues — APPROVED** [review]
How about:

clp-package: Replace Zstandard with Brotli for compressing compression job config. (#323)

**Pull request merged**